specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json

```diff
@@ -2140,6 +2140,57 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{operationalInsightsResourceProvider}/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/entities/{entityId}/timeline": {
+      "post": {
+        "x-ms-examples": {
+          "Entity timeline": {
+            "$ref": "./examples/entities/timeline/PostTimelineEntity.json"
+          }
+        },
+        "tags": [
+          "Entities"
+        ],
+        "description": "Timeline for an entity.",
```

Comment:
>Timeline for an entity. [](start = 24, length = 23)

What does this do to the timeline of an entity? Does it get it? Does it set it? Does it delete it?

Reply:
Ok I saw the example and it looks like you are retrieving the timeline.

1. The name of the POST action should include a verb. If this stays a POST, I would recommend something like `getTimeline` or `listEvents`
2. The name "timeline" is rather vague. It seems like you are retrieving a collection of some sort of events? Why not make it explicit in the name?
3. Why are you using POST for this instead of GET?

---
In reply to: [513849470](https://github.com/Azure/azure-rest-api-specs/pull/11149#discussion_r513849470) [](ancestors = 513849470)

Reply:
1. Rename to "getTimeline". timeline is the scenario name we display in the product. (entities timeline)
2. the reason for POST is that in the future we will add filters (like ids array) that will exceeded the GET url length limit.